src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -1201,5 +1201,15 @@ public static ConditionalAccessExpressionSyntax GetParentConditionalAccessExpres
 
             return null;
         }
+
+        public static ConditionalAccessExpressionSyntax GetInnerMostConditionalAccessExpression(this SyntaxNode node)
+        {
+            if (!(node is ConditionalAccessExpressionSyntax))
+            {
+                return null;
+            }
+
+            return node.DescendantNodesAndSelf().OfType<ConditionalAccessExpressionSyntax>().Last();
```

Comment:
In general, be careful of using Ancestors or Descendants in expression contexts, as you might accidentally cross into a lambda.